proc/threads.c

```diff
@@ -35,11 +36,15 @@ struct {
 	vm_map_t *kmap;
 	spinlock_t spinlock;
 	lock_t lock;
+
+	/* Synchronized by spinlock */
 	thread_t *ready[8];
+	/*
+	 * Write to current[i] only by i'th cpu, with spinlock set.
+	 * Read from current[i] by i'th cpu allowed without spinlock.
+	 */
```

Comment:
Spinlock does nothing if it's used only for writes - let's just assume atomicity (and make these **pointers** volatile)

Reply:
It's not used only for writes - it's used for reads on other cores.
The idea was that the i'th core it's not required because it cannot be writing at the same time as it's reading.